Code/GraphMol/FMCS/FMCS.cpp

```diff
@@ -236,15 +245,54 @@ bool checkAtomChirality(const MCSAtomCompareParameters& p,
   return true;
 }
 
+bool checkAtomDistance(const MCSAtomCompareParameters& p,
+                       const ROMol& mol1, unsigned int atom1,
+                       const ROMol& mol2, unsigned int atom2,
+                       MCSCompareFunctionsData& cfd){
+    unsigned int confId1 = cfd.conformerIdxMap[&mol1];
+    unsigned int confId2 = cfd.conformerIdxMap[&mol2];
+    // If there is no setting for conformer idxs, default to 0
+    if (mol1.getNumConformers() <= confId1 ||
+        mol2.getNumConformers() <= confId2){
+      throw std::runtime_error(
+        "FMCS. Invalid argument. requested conformer unavailable");
+    }
+    const Atom* atm1 = mol1.getAtomWithIdx(atom1);
+    const Atom* atm2 = mol2.getAtomWithIdx(atom2);
+    const Atom* mapAtmFirst = atm1<atm2 ? atm1 : atm2;
+    const Atom* mapAtmSecond = atm1<atm2 ? atm2 : atm1;
+    MCSAtomDistanceCache::iterator cacheIt;
+    std::map<const Atom*, bool>::iterator innerMapIt;
+
+    cacheIt = cfd.atomDistanceCache.find(mapAtmFirst);
```

Comment:
this cache appears to speed up performance a lot. Before introducing it, I was seeing the distance parameter slowing performance by ~60x, then the hit went to ~10x.